chat-plugins/calculator.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+function isNumeric(str) {
+	return !isNaN(parseFloat(str));
+}
+
+function parseMathematicalExpression(infix) {
+	// Shunnting-yard Algorithm -- https://en.wikipedia.org/wiki/Shunting-yard_algorithm
+	let outputQueue = [];
+	let operatorStack = [];
+	let operators = {
+		"^": {
+			precedence: 4,
+			associativity: "Right",
+		},
+		"/": {
+			precedence: 3,
+			associativity: "Left",
+		},
+		"*": {
+			precedence: 3,
+			associativity: "Left",
+		},
+		"+": {
+			precedence: 2,
+			associativity: "Left",
+		},
+		"-": {
+			precedence: 2,
+			associativity: "Left",
+		},
+	};
+	infix = infix.replace(/\s+/g, "");
+	infix = infix.split(/([+\-*/^()])/);
+	infix = infix.filter(token => token);
+	let i;
+	for (i = 0; i < infix.length; i++) {
+		let token = infix[i];
+		if (isNumeric(token) === true) {
+			outputQueue.push(token);
+		} else if ("^*/+-".indexOf(token) !== -1) {
+			let o1 = token;
+			let o2 = operatorStack[operatorStack.length - 1];
+			while ("^*/+-".indexOf(o2) !== -1 && ((operators[o1].associativity === "Left" && operators[o1].precedence <= operators[o2].precedence) || (operators[o1].associativity === "Right" && operators[o1].precedence < operators[o2].precedence))) {
+				outputQueue.push(operatorStack.pop());
+				o2 = operatorStack[operatorStack.length - 1];
+			}
+			operatorStack.push(o1);
+		} else if (token === "(") {
+			operatorStack.push(token);
+		} else if (token === ")") {
+			while (operatorStack[operatorStack.length - 1] !== "(") {
+				outputQueue.push(operatorStack.pop());
+			}
+			operatorStack.pop();
+		}
+	}
+	while (operatorStack.length > 0) {
+		outputQueue.push(operatorStack.pop());
+	}
+	return outputQueue;
+}
+
+function solveRPN(rpn) {
+	let resultStack = [];
+	let i;
+	for (i = 0; i < rpn.length; i++) {
+		if (isNumeric(rpn[i]) === true) {
+			resultStack.push(rpn[i]);
+		} else {
+			let a = resultStack.pop();
```

Comment:
I would convert these to numbers here, for readability of your subsequent stuff.